test/test_datasets_utils.py

```diff
@@ -214,5 +216,30 @@ def test_verify_str_arg(self):
         pytest.raises(ValueError, utils.verify_str_arg, "b", ("a",), "arg")
 
 
+class TestFolderUtils:
+    @pytest.mark.parametrize(
+        ("kwargs", "expected_error_msg"),
+        [
+            (dict(is_valid_file=lambda path: pathlib.Path(path).suffix in {".png", ".jpeg"}), "classes c"),
+            (dict(extensions=".png"), r"classes b, c.*?[.]png"),
+            (dict(extensions=[".png", ".jpeg"]), "c.*?[.]png, [.]jpeg"),
+        ],
+    )
+    def test_make_dataset_no_valid(self, tmpdir, kwargs, expected_error_msg):
+        tmpdir = pathlib.Path(tmpdir)
```

Comment:
TIL, that `tmpdir` is not a `pathlib.Path`, but rather a custom type which acts somewhat like a `pathlib.Path`. Still, methods like `.touch` do not exists, so we cast here.